HIP/hip-583.md

```diff
@@ -10,220 +10,518 @@ status: Accepted
-
-Examples of each format are noted below
-
-| Identifier Type | Example |
-| --- | --- |
-| Hedera Account ID | 0.0.10 |
-| Hedera Long-Form Account ID | 0.0.CIQNOWUYAGBLCCVX2VF75U6JMQDTUDXBOLZ5VJRDEWXQEGTI64DVCGQ ([base32url bytes](https://datatracker.ietf.org/doc/html/rfc4648#section-6)) |
-| Hedera Account Long-Zero address | 0x000000000000000000000000000000000000000a (for accountId 0.0.10) |
-| Ethereum Account Address / public-address | 0xb794f5ea0ba39494ce839613fffba74279579268 |
-
-To provide greater context for this HIP some additional terms are defined here in advance
+![auto-create-with-key.png](../assets/hip-583/auto-create-with-key.png)
 
-- `Ethereum-native wallets` - wallets that utilize the Ethereum JSON RPC APIs to interact with an Ethereum node. Accounts created via this wallet are ECDSA key based and only expose the account `public-address`.
-- `Hedera-native wallets` - wallets that utilize the  Hedera SDK or gRPC HAPI calls to interact with Hedera consensus and mirror nodes. Historically, accounts created via these wallet are `ED25519` based and usually identify the account via the `Hedera account id`.
-- `auto-create` - a unique account creation flow in which the public key that a crytotransfer is addressed to is used to automatically create a new account (if none exists) without an explicit `CryptoCreate` transaction being submitted.
-- `Hollow account` - a partially created account which lacks a public key but does have a valid `shard.rela.num` accountId. Such an account may receive assets but may not take part in transactions that require its signature.
-- `lazy-create` - a unique account creation flow in which the public address (extracted from an ECDSA public key) provided is used to automatically create a hollow account without an explict CryptoCreate transction being submitted.
+Add the `alias` property to the `CryptoCreateTransactionBody` [proto file](https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_create.proto)
 
-Post the implementation of [HIP 32](https://hips.hedera.com/hip/hip-32), only accounts created via `auto-create` using the `CryptoTransfer` transaction have aliases set. Wallets and exchanges often utilize the `CryptoCreate` transaction and therefore have not fully integrated alias logic into their UX flows. This limits full applicability of Smart Contract scenarios for Hedera accounts created using `Hedera-native wallets` and web3 tool compatibility for new users coming from external chains using `Ethereum-native wallets`. 
+```proto
+message CryptoCreateTransactionBody {
+	...
+	/**
+     * The bytes to be used as the account's alias. The bytes must be formatted in one of the following ways:
+     * 
+     * 1. It will be the serialization of a protobuf Key message for an ED25519/ECDSA primitive key type</li>
```

Comment:
This comment should be updated to not include the case for key alias (should be done in the actual protobufs as well).